README.md
# -user-eugeniobinini-files-home-eugeniobinini-mysite
# -user-eugeniobinini-files-home-eugeniobinini-mysite
# -user-eugeniobinini-files-home-eugeniobinini-mysite


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "saint francois de la salle, saint jean baptiste de la salle, saint antoine de padoue"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  pip freeze > requirements.txt
    8  git status
    9  git add .
   10  git commit -m "on acceuile sans tarder notre deuxieme marcheuse, claire. je travaille dans la restoration des vitraux"
   11  pip install Flask-Migrate
   12  export FLASK_APP=flask_app.py
   13  flask db init
   14  flask db migrate
   15  git status
   16  git add .
   17  git commit -m "jean michel, choissisez un troisieme theme, sil vous plait"
   18  pip install Flask-Migrate
   19  export FLASK_APP=flask_app.py
   20  flask db init
   21  flask db migrate
   22  git status
   23  git add .
   24  git commit -m "cadeau pour vous en attendant, pour aprofondir vos connaissances sur la foi chretienne"
   25  flask db stamp head
   26  pip freeze > requirements.txt
   27  git status
   28  git add .
   29  git commit -m "quand jesus a du se quitter d'une taxe"
   30  flask db migrate
   31  flask db upgrade
   32  git status
   33  git add .
   34  git commit -m "on se resource bien sur, mais il faut revenir le 26 juin. rebienvenus chez vous et vous pretendez a ce titre du marcheur"
   35  flask db migrate
   36  flask db upgrade
   37  flask shell
   38  echo "# -user-eugeniobinini-files-home-eugeniobinini-mysite" >> README.md
   39  git init
   40  git add README.md
   41  git commit -m "first commit"
   42  git remote add origin https://github.com/Petersouteneur/-user-eugeniobinini-files-home-eugeniobinini-mysite.git
   43  git push -u origin master
   44  echo "# -user-eugeniobinini-files-home-eugeniobinini-mysite" >> README.md
   45  git init
   46  git add README.md
   47  git commit -m "first commit"
   48  git remote add origin https://github.com/Petersouteneur/-user-eugeniobinini-files-home-eugeniobinini-mysite.git
   49  git push -u origin master
   50  echo "# -user-eugeniobinini-files-home-eugeniobinini-mysite" >> README.md
   51  git init
   52  git add README.md
   53  git commit -m "first commit"
   54  git remote add origin https://github.com/Petersouteneur/-user-eugeniobinini-files-home-eugeniobinini-mysite.git
   55  git push -u origin master
   56  history
   57  history < history.txt
   58  cat < history.txt
   59  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="eugeniobinini",
    password="Exeter22",
    hostname="eugeniobinini.mysql.pythonanywhere-services.com",
    databasename="eugeniobinini$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "456546545689789789321321321"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/a3c52a13058e_.py
"""empty message

Revision ID: a3c52a13058e
Revises: 
Create Date: 2018-08-21 22:45:55.695413

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'a3c52a13058e'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/dfe18b4bf631_.py
"""empty message

Revision ID: dfe18b4bf631
Revises: a3c52a13058e
Create Date: 2018-08-21 22:52:14.944256

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'dfe18b4bf631'
down_revision = 'a3c52a13058e'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
